src/app/shared/pipes/order-by.pipe.ts

```diff
@@ -0,0 +1,37 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import Course from '../../pages/courses-page/components/course/course.types';
+
+@Pipe({
+  name: 'orderBy',
+})
+export class OrderByPipe implements PipeTransform {
+
+  transform(array: Array<Course>, order = ''): any[] {
+    if (!Array.isArray(array)) {
+      return;
+    }
+    array.sort((a: any, b: any) => {
```

Comment:
I guess this code is easer to read and support: 
`const ascSorterFn = (a, b): number => new Date(a[sortBy]).getTime() - new Date(b[sortBy]).getTime();`
`const descSorterFn = (a, b): number => new Date(b[sortBy]).getTime() - new Date(a[sortBy]).getTime();`

    return array.sort(sortOrder === 'ASC' ? ascSorterFn : descSorterFn);`

Just use enum instead of 'ASC'

Reply:
done